UI/EncyclopediaDetailPanel.cpp

```diff
@@ -949,9 +985,11 @@ void EncyclopediaDetailPanel::HandleSearchTextEntered() {
     std::set<std::pair<std::string, std::string>> already_listed_results;  // pair of category / article name
 
     // assemble link text to all pedia entries, indexed by name
-    std::vector<std::string> dir_names = GetSearchTextDirNames();
-    for (const std::map<std::string, std::vector<EncyclopediaArticle>>::value_type& article : GetEncyclopedia().articles) {
-        dir_names.push_back(article.first);
+    std::unordered_set<std::string> dir_names;
```

Comment:
nevermind... the outer loop is over the result of `GetSearchTextDirNames()` not over `dir_names`.